js/components/plates.js

```diff
@@ -356,6 +388,17 @@ export default class Plates extends PureComponent {
       const type = event.data.type
       if (type === 'output') {
         this.handleDataFromWorker(event.data.data)
+      } else if (type === 'saveModel') {
+        saveModelToCloud(event.data.data.savedModel,
+        function () {
+          this.setState({ savingModel: true })
+        }.bind(this),
+        function (modelId) {
+          this.setState({
+            lastStoredModel: modelId,
+            savingModel: false
+          })
+        }.bind(this))
```

Comment:
I'd remove start callback and just call `this.setState({ savingModel: true })` after or before calling `saveModelToCloud`. startCallback is synchronous, so it doesn't really need to be a callback.
Btw, `function () { }.bind(this)` can be replaced by `() => { }`

Reply:
d'oh, good point, that makes much more sense